staging/src/k8s.io/apiserver/pkg/server/filters/waitgroup.go

```diff
@@ -38,7 +42,10 @@ func WithWaitGroup(handler http.Handler, longRunning apirequest.LongRunningReque
 
 		if !longRunning(req, requestInfo) {
 			if err := wg.Add(1); err != nil {
-				http.Error(w, "apiserver is shutting down.", http.StatusInternalServerError)
+				// When apiserver is shutting down, signal clients to retry
+				w.Header().Add("Retry-After", "1")
+				statusErr := apierrors.NewServiceUnavailable("apiserver is shutting down").Status()
+				http.Error(w, runtime.EncodeOrDie(scheme.Codecs.LegacyCodec(v1.SchemeGroupVersion), &statusErr), int(statusErr.Code))
```

Comment:
does this write the correct Content-Type header? I think this will send json content with a text/plain header

Reply:
good catch

Reply:
http.Error stomps content-type to text/plain

Reply:
fixed.